server/src/main/java/org/apache/cloudstack/direct/download/DirectDownloadManagerImpl.java

```diff
@@ -320,14 +325,38 @@ private void logUsageEvent(VMTemplateVO template, long poolId) {
      */
     private DirectDownloadCommand getDirectDownloadCommandFromProtocol(DownloadProtocol protocol, String url, Long templateId, PrimaryDataStoreTO destPool,
                                                                        String checksum, Map<String, String> httpHeaders) {
+        int connectTimeout = DEFAULT_DIRECT_DOWNLOAD_CONNECT_TIMEOUT;
+        int soTimeout = DEFAULT_DIRECT_DOWNLOAD_SOCKET_TIMEOUT;
+        int connectionRequestTimeout = DEFAULT_DIRECT_DOWNLOAD_CONNECTION_REQUEST_TIMEOUT;
+        if (DownloadProtocol.HTTP.equals(protocol) ||
+                DownloadProtocol.HTTPS.equals(protocol) ||
+                DownloadProtocol.METALINK.equals(protocol)) {
+            try {
+                connectTimeout = Integer.parseInt(configDao.getValue(DirectDownloadConnectTimeout.key()));
```

Comment:
Please do not use this and use the config key value() method

Reply:
Done